solr/core/src/java/org/apache/solr/handler/admin/api/RestoreCollectionAPI.java

```diff
@@ -70,15 +69,17 @@
 @Path("/backups/{backupName}/restore")
 public class RestoreCollectionAPI extends BackupAPIBase {
 
-  private static final Set<String> CREATE_PARAM_ALLOWLIST =
-      Set.of(
-          COLL_CONF,
-          REPLICATION_FACTOR,
-          NRT_REPLICAS,
-          TLOG_REPLICAS,
-          PULL_REPLICAS,
-          CREATE_NODE_SET_PARAM,
-          CREATE_NODE_SET_SHUFFLE_PARAM);
+  private static final Set<String> CREATE_PARAM_ALLOWLIST = makeCreateParamAllowList();
+
+  private static Set<String> makeCreateParamAllowList() {
```

Comment:
Just playing around here... consider this code:
```
Stream.concat(
              CollectionHandlingUtils.numReplicasProperties().stream(),
              Stream.of(COLL_CONF, REPLICATION_FACTOR, CREATE_NODE_SET_PARAM,
          CREATE_NODE_SET_SHUFFLE_PARAM))
          .collect(Collectors.toUnmodifiableSet());
```

Reply:
Good suggestion, thank you.